src/lib/Microsoft.Fx.Portability/Analyzer/PublicKeyToken.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Fx.Portability
+{
+    public readonly struct PublicKeyToken : IEquatable<PublicKeyToken>
+    {
+        private readonly ImmutableArray<byte> _token;
+
+        public PublicKeyToken(ImmutableArray<byte> bytes)
+        {
+            _token = bytes;
+        }
+
+        public ImmutableArray<byte> Token => _token.IsDefault ? ImmutableArray<byte>.Empty : _token;
+
+        public bool Equals(PublicKeyToken other)
+        {
+            if (Token.Length != other.Token.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < Token.Length; i++)
+            {
+                if (Token[i] != other.Token[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public override bool Equals(object obj) => obj is PublicKeyToken other ? Equals(other) : false;
+
+        public override int GetHashCode()
+        {
+            int hash = 19;
+            unchecked
+            {
+                foreach (var item in Token)
+                {
+                    hash = (hash * 31) + item;
+                }
+            }
```

Comment:
It's just a standard hashcode generation - I'd like to use System.HashCode, but we're not targeting .NET Core

Reply:
What about using something like system.security.cryptography.sha256?

Reply:
That's overkill for GetHashCode. This is a standard implementation similar to https://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode. We don't need cryptographic-level hashing - just something to get a best-effort distribution. Using primes for the seed and the multiple help with that.

Reply:
Using system.security.cryptography.sha256 feels overkill as well since this is not for security purpose.   Just rather using standard library instead of creating our own. Looks like many people used https://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode. I'm fine with it.